checker/src/org/checkerframework/checker/index/samelen/SameLenTransfer.java

```diff
@@ -42,32 +46,33 @@ public SameLenTransfer(CFAnalysis analysis) {
             AssignmentNode node, TransferInput<CFValue, CFStore> in) {
         TransferResult<CFValue, CFStore> result = super.visitAssignment(node, in);
 
-        // Check if an array is being created.
+        // Handle "new T[a.length]".
 
         if (node.getExpression() instanceof ArrayCreationNode) {
             ArrayCreationNode acNode = (ArrayCreationNode) node.getExpression();
             if (acNode.getDimensions().size() == 1 && isArrayLengthAccess(acNode.getDimension(0))) {
-                // This array that's being created is the same size as the other one.
+                // node is known to be "new T[a.length]"
                 FieldAccessNode arrayLengthNode = (FieldAccessNode) acNode.getDimension(0);
                 Node arrayLengthNodeReceiver = arrayLengthNode.getReceiver();
+                // arrayLengthNode is known to be "new T[arrayLengthNodeReceiver.length]"
 
                 Receiver targetRec =
                         FlowExpressions.internalReprOf(analysis.getTypeFactory(), node.getTarget());
                 Receiver otherRec =
                         FlowExpressions.internalReprOf(
                                 analysis.getTypeFactory(), arrayLengthNodeReceiver);
 
-                AnnotationMirror arrayLengthAnnotation =
+                AnnotationMirror arrayLengthNodeAnnotation =
                         aTypeFactory
                                 .getAnnotatedType(arrayLengthNodeReceiver.getTree())
                                 .getAnnotationInHierarchy(UNKNOWN);
 
                 AnnotationMirror combinedSameLen =
                         aTypeFactory.createCombinedSameLen(
-                                otherRec.toString(),
                                 targetRec.toString(),
```

Comment:
Target is the lefthand side of the assignment, so if the assignment is `a = new T[b.length]` then ` targetRec.toString()` is `a`.  I updated the comments to make this clear.